### ISSUE TITLE
Seeking Opinions : Separating general functionalities in the geometry module

### ISSUE THREAD

**Issue by anandhupvr** [kornia/kornia#2460]
## 🚀 Feature

I would like to initiate a discussion and gather opinions from the maintainers and the community regarding the idea of separating general functionalities in the geometry module.
I propose separating certain general functionalities within the geometry module into separate files for improved organization and maintainability. Currently, I have noticed that polynomial solvers are within the `kornia.geometry.epipolar.fundamental.py` file. However, separating these polynomial solver functions into a dedicated file named `polynomial_solver.py` located in the same directory
## Motivation

[opencv-polynomial-solver](https://github.com/opencv/opencv/blob/4.x/modules/calib3d/src/polynom_solver.h)
## Pitch

```
   kornia.geometry.epipolar.polynomial_solver.py
    
         def solve_quadratic(coeffs: Tensor) -> Tensor:

          
          def solve_cubic(coeffs: Tensor) -> Tensor:

          
          def solve_quartic(coeffs: Tensor) -> Tensor:

```
or with this naming 
```
         def solve_deg2(coeffs: Tensor) -> Tensor:

          
          def solve_deg3(coeffs: Tensor) -> Tensor:

          
          def solve_deg4(coeffs: Tensor) -> Tensor:

```
## Request for Opinions:

1.  Do you think it would be beneficial to separate the polynomial solver functions into their own file?
2. How would separating the polynomial solver functions impact the overall structure and organization of the geometry module?
3. Are there any alternative suggestions or considerations that should be taken into account?

## Additional context

______________________________________________________________________

#### Consider also to contribute to Kornia universe projects :)

<sub>

- [**Tutorials**](https://github.com/kornia/tutorials): our repository containing the tutorials.

</sub>

**Comment by edgarriba** [comment]
I like the idea -- and I'd even go to an upper level like `kornia.geometry.solvers.polynomial_solver.py`

**Comment by edgarriba** [comment]
@ducha-aiki @johnnv1 @cjpurackal

**Comment by cjpurackal** [comment]
sounds good! Perhaps we can move some functions from `kornia.utils.helpers`
https://github.com/kornia/kornia/blob/40376f26dff93b6b251415f2aeb9750466c5c9e4/kornia/utils/helpers.py#L213
https://github.com/kornia/kornia/blob/40376f26dff93b6b251415f2aeb9750466c5c9e4/kornia/utils/helpers.py#L228
etc (used in `kornia.geometry.homography` etc) to `kornia.geometry.solvers` .

**Comment by anandhupvr** [comment]
also adding Gauss-Newton and Levenberg-Marquardt (Nonlinear solvers) solvers to Kornia's geometry module would benefit SFM and enhance functionality.

**Comment by cjpurackal** [comment]
Yes, non-linear optimizers are on track for our road map this year. We are looking into a couple of libs like https://github.com/facebookresearch/theseus and https://github.com/pypose/pypose for possible integration. 
I'm currently working on this, happy to share more details once I put out something in a couple of weeks.
